File: app/src/main/java/com/example/okhttp/Activity/ThirdActivity.java
package com.example.okhttp.Activity;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.AppCompatSpinner;
import android.support.v7.widget.DefaultItemAnimator;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.Toolbar;
import android.view.MenuItem;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import com.example.okhttp.Adapter.Mydapter;
import com.example.okhttp.R;
import com.example.okhttp.tool.OkHttpRequest;

import java.util.ArrayList;


public class ThirdActivity extends AppCompatActivity implements AdapterView.OnItemSelectedListener{

    private Toolbar toolbar;
    private RecyclerView scorerecyclerView;
    private Mydapter mydapter;
    private AppCompatSpinner spinner;

    private ArrayList<String> strings;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_third);

        initView();
    }
    public void initView(){
        toolbar = findViewById(R.id.thirdtoobar);
        setSupportActionBar(toolbar);
        toolbar.setSubtitle("成绩查询");

        getSupportActionBar().setDisplayHomeAsUpEnabled(true);


        TextView title = findViewById(R.id.title);
        title.setText(OkHttpRequest.getStudentName()+",你好！");

        scorerecyclerView = findViewById(R.id.allscore_RecycleView);
        scorerecyclerView.setLayoutManager(new LinearLayoutManager(this));
        scorerecyclerView.setItemAnimator(new DefaultItemAnimator());
        mydapter = new Mydapter(this);
        scorerecyclerView.setAdapter(mydapter);

        strings = new ArrayList<>();
        strings.add("历年成绩");
        for(ArrayList<String> list: OkHttpRequest.getScore_inquiryList()){
            String s = list.get(0) +"学年度第"+ list.get(1)+"学期";
            if(strings.size()>0 && !strings.get(strings.size()-1).equals(s)){
                strings.add(s);
            }else if(strings.size() == 0){
                strings.add(s);
            }
        }
        spinner = findViewById(R.id.thirdSpinner);
        ArrayAdapter<String> adapter = new ArrayAdapter<>(this,
                android.R.layout.simple_spinner_item,strings);
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
        spinner.setAdapter(adapter);
        spinner.setSelection(0);
        spinner.setOnItemSelectedListener(this);
        spinner.setVisibility(View.VISIBLE);

    }

    @Override
    public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
        String s = strings.get(position);
        if(s.equals("历年成绩")){
           mydapter.setIsShowAllscore(true,null);
        }else {
            mydapter.setIsShowAllscore(false,s);
        }
        mydapter.notifyDataSetChanged();
    }

    @Override
    public void onNothingSelected(AdapterView<?> parent) {

    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if(item.getItemId() == android.R.id.home){
            finish();
            return true;
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/example/okhttp/Activity/ThirdActivity/ThirdActivity.kt
package com.example.okhttp.Activity.ThirdActivity

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v7.widget.AppCompatSpinner
import android.support.v7.widget.DefaultItemAnimator
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.support.v7.widget.Toolbar
import android.view.MenuItem
import android.view.View
import android.widget.AdapterView
import android.widget.ArrayAdapter
import android.widget.TextView

import com.example.okhttp.Adapter.Mydapter
import com.example.okhttp.R

import kotlin.collections.ArrayList


class ThirdActivity : AppCompatActivity(), IThirdVIew,AdapterView.OnItemSelectedListener {

    private var toolbar: Toolbar? = null
    private var scorerecyclerView: RecyclerView? = null
    private var spinner: AppCompatSpinner? = null
    private var strings: ArrayList<String>? = null

    private var mydapter: Mydapter? = null
    private var title:TextView? = null

    private var presenter:IThirdPresenter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_third)
        initView()
        presenter!!.changeTitle()
        initspinnerData()
    }

    fun initView(){
        toolbar = findViewById(R.id.thirdtoobar)
        setSupportActionBar(toolbar)

        toolbar!!.subtitle = "成绩查询"
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)

        spinner = findViewById(R.id.thirdSpinner)

        title = findViewById<TextView>(R.id.title)
        scorerecyclerView = findViewById(R.id.allscore_RecycleView)
        scorerecyclerView!!.layoutManager = LinearLayoutManager(this)
        scorerecyclerView!!.itemAnimator = DefaultItemAnimator()
        mydapter = Mydapter(this)
        scorerecyclerView!!.adapter = mydapter

        presenter = ThirdPresenter(this)
    }

    override fun changeTitle(studentname:String) {
        title!!.text = studentname + ",你好！"
    }

    override fun initspinnerData(){

        strings = presenter!!.initspinnerData()

        val adapter = ArrayAdapter(this,
                android.R.layout.simple_spinner_item, strings)
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
        spinner!!.adapter = adapter
        spinner!!.setSelection(0)
        spinner!!.onItemSelectedListener = this
        spinner!!.visibility = View.VISIBLE

    }

    override fun onItemSelected(parent: AdapterView<*>, view: View, position: Int, id: Long) {
        val s = strings!![position]
        if (s == "历年成绩") {
            mydapter!!.setIsShowAllscore(true, "")
        } else {
            mydapter!!.setIsShowAllscore(false, s)
        }
        mydapter!!.notifyDataSetChanged()
    }

    override fun onNothingSelected(parent: AdapterView<*>) {
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            finish()
            return true
        }
        return super.onOptionsItemSelected(item)
    }
}
